draft-ietf-tls-esni.md

```diff
@@ -216,64 +239,83 @@ ECHConfigs structure.
     ECHConfig ECHConfigs<1..2^16-1>;
 ~~~~
 
-The ECHConfigs structure contains one or more ECHConfig structures in decreasing
-order of preference. This allows a server to support multiple versions of ECH
-and multiple sets of ECH parameters.
+The `ECHConfigs` structure contains one or more `ECHConfig` structures in
+decreasing order of preference. This allows a server to support multiple
+versions of ECH and multiple sets of ECH parameters.
 
-The ECHConfig structure contains the following fields:
+The `ECHConfig` structure contains the following fields:
 
 version
-: The version of the structure. For this specification, that value SHALL be
-0xff07. Clients MUST ignore any ECHConfig structure with a version they do not
-understand.
+: The version of ECH for which this configuration is used. Versioning is
+described below in {{versioning}}. Clients MUST ignore any `ECHConfig` structure
+with a version they do not support.
+
+length
+: The length, in bytes, of the next field.
 
 contents
 : An opaque byte string whose contents depend on the version of the structure.
-For this specification, the contents are an ECHConfigContents structure.
+For this specification, the contents are an `ECHConfigContents` structure.
+[[TODO: `contents` isn't a named field in the `ECHConfig` structure. I'm
```

Comment:
That's true. We can rewrite this as a paragraph, e.g., saying "The remaining contents of the `ECHConfig` structure are an opaque byte string..."

Reply:
Hmmm, I'm curious though: shouldn't this field have a label?
https://tools.ietf.org/html/rfc8446#section-3.8

Reply:
It's a union, basically, whose value depends on the type. I don't know what would be the desired label, so I'd be fine with the current text or what's proposed above.

Reply:
What I don't like about your proposal is that the `length` field gives the length of a nameless field. I think `content` would be a reasonable label.

Reply:
I see little difference between that and `contents`, so, let's just keep the current text?

Reply:
Oh, "contents" is fine. The suggested change is to give the variant a label.

Reply:
Hmm, I'm not sure what to here. I'm resolving this and leaving the text as is.